virtool/tasks/runner.py

```diff
@@ -14,29 +13,48 @@
 
 class TaskRunner:
     def __init__(
-        self, data: DataLayer, tasks_client: AbstractTasksClient, app: Application
+        self, tasks_client: AbstractTasksClient, app: Application
     ):
-        self._data = data
         self._tasks_client = tasks_client
         self.app = app
+        self._current_task: Task = None
 
     async def run(self):
-        logging.info("Started task runner")
-
         try:
             while True:
                 logging.info("Waiting for next task")
-
                 task_id = await self._tasks_client.pop()
 
                 await self.run_task(task_id)
 
+                await asyncio.shield(self._current_task)
```

Comment:
I think this shield is redundant as the underlying is already shielding the task we want to let keep running